src/app.js

```diff
@@ -6,8 +6,225 @@ import _ from 'underscore';
+const showTripDetails = function showTripDetails(trip){
+  const individualTripListElement = $('#individual-trip-detail');
+  individualTripListElement.html('');
+
+  const generatedHTMLTripDetails = individualTripTemplate(trip.attributes);
+
+  individualTripListElement.append(generatedHTMLTripDetails);
+  $('#individual-trip-detail').show();
+
+  //////////////////////////
+  // Reservation Details ///
+  //////////////////////////
+  const reserveTrip = (event) => {
+    event.preventDefault();
+    console.log('in reserveTrip');
+    const RES_FIELDS = ['trip_id', 'name', 'age', 'email'];
+    let reservationData = {}
+    RES_FIELDS.forEach((field) => {
+      const input = $(`#add-reservation input[name="${ field }"]`);
+      const val = input.val();
+      if (val != '') {
+        reservationData[field] = val;
+      }
+      input.val('');
+    })
+    const reservation = new Reservation(reservationData);
+    console.log(reservationData);
+    console.log(reservation);
+    if (!reservation.isValid()) {
+      handleValidationFailures(reservation.validationError);
+      return;
+    }
+    reservation.save({}, {
+      success: (model, response) => {
+        console.log('success');
+        reportStatus('success', 'Trip reserved!');
+        $('#add-reservation').hide();
+      },
+      error: (model, response) => {
+        console.log(response);
+        console.log(response.responseJSON["errors"]);
+        reportStatus('error', 'Not reserved!')
+        handleValidationFailures(response.responseJSON["errors"]);
+      },
+    });
+  };
+  //////////////////////////
+  // reservation handler ///
+  //////////////////////////
+  const reserveTemplateShow = function reserveTemplateShow(trip){
+    const reservationElement = $('#reservation-form');
+    reservationElement.html('');
+    const generatedHTMLreserve = reserveTemplate(trip.attributes);
+    console.log(trip.attributes);
+    reservationElement.append(generatedHTMLreserve);
+  };
+
+  $('#reserve').show();
+  $('#reserve').on('click', function(event) {
+    reserveTemplateShow(trip);
+    $('#reserve').hide();
+    $('#add-reservation').on('submit', function(event) {
+      event.preventDefault();
+      console.log('on submit');
+      reserveTrip(event);
+    });
+
+  });
+};
+
+// pulled out fetchTripDetails function above
+const fetchTripDetails = function fetchTripDetails(event) {
+  $('#add-reservation').hide();
+  let tripId = $(this).attr('data-id');
+  let trip = tripList.get(tripId)
+  trip.fetch({
+    success: () => {
+      showTripDetails(trip);
+    }
+  });
+};
+
+//////////////////////////
+// sorting ///////////////
+//////////////////////////
+const render = function render(tripList) {
+  const tripListElement = $('#trip-list');
+  tripListElement.empty();
+
+  tripList.forEach((trip) => {
+    const generatedHTML = tripTemplate(trip.attributes);
+    tripListElement.append($(generatedHTML));
+
+  });
+  $('th.sort').removeClass('current-sort-field');
```

Comment:
Interesting how you're handling the styling for the sort field here.  It works well.